src/mono/mono/metadata/image-internals.h

```diff
@@ -19,6 +30,6 @@ MonoImage*
 mono_image_load_module_checked (MonoImage *image, int idx, MonoError *error);
 
 MonoImage *
-mono_image_open_a_lot (MonoAssemblyLoadContext *alc, const char *fname, MonoImageOpenStatus *status);
+mono_image_open_a_lot (MonoAssemblyLoadContext *alc, const char *fname, MonoImageOpenStatus *status, gboolean not_executable);
```

Comment:
Let's push this a bit more:
```suggestion
mono_image_open_a_lot (MonoAssemblyLoadContext *alc, const char *fname, MonoImageOpenStatus *status, const MonoImageOpenOptions *options);
```

And make the callers of `mono_image_open_a_lot` set the options the way that is expected

Reply:
Thanks!